src/plugins/interactive-media/interactive-media-editor.js

```diff
@@ -1,102 +1,111 @@
 import by from 'thenby';
-import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+import reactPlayerNs from 'react-player';
 import { useTranslation } from 'react-i18next';
+import uniqueId from '../../utils/unique-id.js';
 import validation from '../../ui/validation.js';
-import { Form, Input, Radio, Switch } from 'antd';
 import Timeline from '../../components/timeline.js';
+import { Form, Input, Radio, Spin, Switch } from 'antd';
 import { MEDIA_TYPE, SOURCE_TYPE } from './constants.js';
 import { removeItemAt } from '../../utils/array-utils.js';
+import React, { Fragment, useRef, useState } from 'react';
 import ClientConfig from '../../bootstrap/client-config.js';
 import { useService } from '../../components/container-context.js';
 import { sectionEditorProps } from '../../ui/default-prop-types.js';
 import StorageFilePicker from '../../components/storage-file-picker.js';
 import ObjectMaxWidthSlider from '../../components/object-max-width-slider.js';
 
+const ReactPlayer = reactPlayerNs.default || reactPlayerNs;
+
 const FormItem = Form.Item;
 const RadioGroup = Radio.Group;
 const TextArea = Input.TextArea;
 const RadioButton = Radio.Button;
 
 const videoTypes = ['mp4', 'mov', 'avi', 'mkv'];
 const audioTypes = ['mp3', 'flac', 'aac', 'wav'];
+const supportedAspectRatios = [{ h: 16, v: 9 }, { h: 4, v: 3 }];
 
-const ensurePartsOrder = parts => parts.sort(by(part => part.startTimecode));
-const getAspectRatioText = aspectRatio => `${aspectRatio.h}:${aspectRatio.v}`;
-
-const getMediaType = path => {
-  const sanitizedPath = (path || '').trim();
-  const extensionMatches = sanitizedPath.match(/\.([0-9a-z]+)$/i);
-  const extension = extensionMatches && extensionMatches[1];
-
-  if (!extension) {
-    return MEDIA_TYPE.none;
-  }
-  if (audioTypes.includes(extension)) {
-    return MEDIA_TYPE.audio;
-  }
-  if (videoTypes.includes(extension)) {
-    return MEDIA_TYPE.video;
-  }
-  return MEDIA_TYPE.unknown;
+const formItemLayout = {
+  labelCol: { span: 4 },
+  wrapperCol: { span: 14 }
 };
 
 function InteractiveMediaEditor({ content, onContentChanged, publicStorage, privateStorage }) {
   const clientConfig = useService(ClientConfig);
+  const hiddenPlayerContainerRef = useRef(null);
   const { t } = useTranslation('interactiveMedia');
 
-  const { sourceType, sourceUrl, text, width, aspectRatio, showVideo } = content;
+  const getAspectRatioText = givenAspectRatio => `${givenAspectRatio.h}:${givenAspectRatio.v}`;
+  const ensureChaptersOrder = chapters => chapters.sort(by(chapter => chapter.startTimecode));
+  const getDefaultChapter = () => ({ key: uniqueId.create(), title: t('defaultChapterTitle'), startTimecode: 0 });
```

Comment:
I think we should move this function into `interactive-media-info.js` (and also use it there when generating the whole default content).

Reply:
Good idea, moved it there.